From c75200676abdab578e7ce843bbb9e90816880605 Mon Sep 17 00:00:00 2001
From: darkyzhou <darkyoooooo@gmail.com>
Date: Sun, 18 May 2025 11:54:28 +0800
Subject: [PATCH 80/81] cpufreq: adapt phyt_cpufreq_cpu_exit and
 phyt_pstate_cpu_exit to new signature of cpufreq_driver

Signed-off-by: darkyzhou <me@zqy.io>
---
 drivers/cpufreq/phytium_pstate.c | 7 ++-----
 1 file changed, 2 insertions(+), 5 deletions(-)

diff --git a/drivers/cpufreq/phytium_pstate.c b/drivers/cpufreq/phytium_pstate.c
index 54a6a9731..8d01a621b 100644
--- a/drivers/cpufreq/phytium_pstate.c
+++ b/drivers/cpufreq/phytium_pstate.c
@@ -417,13 +417,11 @@ static int phyt_cpufreq_cpu_init(struct cpufreq_policy *policy)
 	return ret;
 }
 
-static int phyt_cpufreq_cpu_exit(struct cpufreq_policy *policy)
+static void phyt_cpufreq_cpu_exit(struct cpufreq_policy *policy)
 {
 	struct phyt_cpudata *cpudata = policy->driver_data;
 
 	kfree(cpudata);
-
-	return 0;
 }
 
 /* Sysfs attributes */
@@ -685,10 +683,9 @@ static int phyt_pstate_cpu_init(struct cpufreq_policy *policy)
 	return ret;
 }
 
-static int phyt_pstate_cpu_exit(struct cpufreq_policy *policy)
+static void phyt_pstate_cpu_exit(struct cpufreq_policy *policy)
 {
 	pr_debug("CPU %d exiting\n", policy->cpu);
-	return 0;
 }
 
 static void phyt_pstate_update_limit(struct cpufreq_policy *policy)
-- 
2.48.1

